### ISSUE TITLE
ClientWebSocket ReceiveAsync infinite await

### ISSUE THREAD

**Issue by zxcvqwerasdf** [dotnet/runtime#110496]
### Description

The problem occurs when the connection at the operation system level has been closed (TCP), but the call to await ReceiveAsync continues to wait, without errors, without a return value. It's also not 100% guarantee reproducible, but sometimes this happening.

### Reproduction Steps

Create simple websocket and connect to any wss server
```
 _webSocket = new ClientWebSocket();
 _webSocket.Options.KeepAliveInterval = TimeSpan.FromSeconds(60);
 _webSocket.Options.KeepAliveTimeout = TimeSpan.FromHours(10);
await _webSocket.ConnectAsync(new Uri("wss://myserver.com"));
while(true)
{
  try
  {
   var rcvBuffer = new byte[2048].AsMemory();
   var readed = await _webSocket.ReceiveAsync(rcvBuffer);
   }
   catch (Exception ex)
   {
       Console.WriteLine(ex.Message);
   }
}
```
and just wait now, there is chance that TCP connection will be closed, but you will continue to wait on this line
```
   var readed = await _webSocket.ReceiveAsync(rcvBuffer);
``` 

### Expected behavior

Return value / throw exception

### Actual behavior

Nothing happening

### Known Workarounds

Create CancellationToken, pass it to ReceiveAsync, create another Task that will check in loop webSocket status, then cancel cancellation token if status != Open

### Configuration

NET 9
Win 10

**Comment by BrennanConroy** [comment]
Set `_webSocket.Options.KeepAliveTimeout = TimeSpan.FromHours(10);` to a lower value. One reason the keep alive options exists is to solve issues with the TCP connection being closed ungracefully and the OS not notifying the connection for many hours.

**Comment by stephentoub** [comment]
>  there is chance that TCP connection will be closed

Closed how?

If it's closed gracefully, the client should absolutely be notified and the receive should not be hanging. If that's what you're observing, we'd need a repro.

If it's closed non-gracefully, e.g. the remote server crashes, then it's normal at the TCP level that a client just waiting to receive data may not ever be notified. This is why keep-alives exist, either at the TCP level or at the web socket level.

**Comment by zxcvqwerasdf** [comment]
> > there is chance that TCP connection will be closed
> 
> Closed how?
> 
> If it's closed gracefully, the client should absolutely be notified and the receive should not be hanging. If that's what you're observing, we'd need a repro.
> 
> If it's closed non-gracefully, e.g. the remote server crashes, then it's normal at the TCP level that a client just waiting to receive data may not ever be notified. This is why keep-alives exist, either at the TCP level or at the web socket level.

I think this happening when it's not closed properly.
It turns out that websocket connection is guaranteed only via ping?
If a websocket does not send ping or other messages at all, is there no guarantee that the websocket maintains an active TCP connection?

**Comment by MihaZupan** [comment]
Closing this one as answered, a keep-alive or some other timeout mechanism (e.g. `CancellationToken`s) is required to reliably observe such connection failures.

Let us know if something wasn't clear.
